src/caches/basic_method_caches.jl

```diff
@@ -116,3 +116,49 @@ function alg_cache(alg::RKMilCommute,prob,u,ΔW,ΔZ,p,rate_prototype,noise_rate_
   Kj = zero(u); Dgj = zero(noise_rate_prototype)
   RKMilCommuteCache(u,uprev,du1,du2,K,gtmp,L,WikJ,Dg,mil_correction,Kj,Dgj,tmp)
 end
+
+struct RKMil_GeneralConstantCache{WikType} <: StochasticDiffEqConstantCache
+  WikJ::WikType
+end
+
+@cache struct RKMil_GeneralCache{uType, rateType, rateNoiseType, WikType} <: StochasticDiffEqMutableCache
+  u::uType
+  uprev::uType
+  tmp::uType
+  du₁::rateType
+  du₂::rateType
+  K::uType
+  L::rateNoiseType
+  mil_correction::uType
+  ggprime::rateNoiseType
+  WikJ::WikType
+end
+
+function alg_cache(alg::RKMil_General,prob,u,ΔW,ΔZ,p,rate_prototype,noise_rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,f,t,dt,::Type{Val{false}})
+  if typeof(ΔW) <: Number || is_diagonal_noise(prob)
+    WikJ = fill_WikJDiagonal_oop(ΔW)
+  elseif alg.is_commutative
```

Comment:
since there is a type change here, this is something we want to resolve at compile time.

Reply:
Then we'll have to write different functions for cache initializations.

Reply:
no you won't

Reply:
How can we resolve this in compile-time without making separate functions?

Reply:
make it type-based, like `Val(true)` or something.